src/gc-arena/src/types.rs

```diff
@@ -26,13 +26,15 @@ impl GcBox {
     }
 
     /// Gets an erased pointer to the value stored inside this box.
+    /// `T` must be the same type that was used with `erase`, so that
+    /// we can correctly compute the field offset.
     #[inline(always)]
-    fn erased_value(&self) -> *mut () {
+    fn erased_value<T>(&self) -> *mut T {
```

Comment:
If this is generic, it's not really 'erased' anymore; calling it `unerase_value` seems better.

Reply:
I've renamed it to `unerased_value`